app/Http/Controllers/Frontend/PirepController.php

```diff
@@ -491,6 +496,9 @@ public function update($id, UpdatePirepRequest $request)
             Flash::error('Pirep not found');
             return redirect(route('admin.pireps.index'));
         }
+        if ($pirep->user_id !== Auth::id()) {
```

Comment:
Good call on this... I thought I had put them in the `UpdatePirepRequest` classes but apparently, I only did that for the ACARS requests. So I think this should actually go into `App/Http/Requests/UpdatePirepRequest`, there should be a 'authorize()` method. Take a look at `App/Http/Requests/Acars/EventRequest`. 

I think something like

```php
$pirepRepo = app(PirepRepository::class); // import it
$pirep = $pirepRepo->find($this->route('pirep_id'));
return $pirep->user_id === Auth::id();
```

The other ones are fine, though. Thanks!

Reply:
Thank you for pointing me there. Those request classes were new to me. I fully agree